### PR TITLE
Added casts related to use of unsafe.Pointer

### PR DESCRIPTION
**Pull request by feyeleanor**
The code now compiles against Go 1.4.1.

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -105,7 +106,7 @@ func (s *Store) getOrCreateTypedKey(k Key, v Value, kt KeyType) *BRecord {
 					}
 				}
 			}
-			br = thing.(*BRecord)
+			br = (*BRecord)(thing)
```
Comment (narula) [inline store.go:109]: What version of go are you using?  I'm building from the tip of devel, and compiling fails with:

./store.go:109: cannot convert thing (type gotomic.Thing) to type *BRecord: need type assertion

Comment (narula) [inline store.go:109]: And this happens with 1.4.1 as well.

**Comment by narula** [comment]
Nevermind -- I was using an old library.  Thanks, merged!

**Comment by feyeleanor** [comment]
I'm interested in using phase reconciliation for a distributed problem I'm working on but not being a specialist in this field I'm somewhat lost digging through the code. Are there any resources other than your paper which you'd recommend for getting to grips with these ideas?

File: store.go (hunk 1 above)
Comment (feyeleanor) [inline store.go:109]: Sorry, didn't mean to seem rude. I was busy with business fluff this week so didn't see your comments until you'd already solved the problem.

**Comment by narula** [comment]
Sadly, no.  However, I'm happy to answer any questions you might have.  Feel free to email me at narula@gmail.com with what you're trying to do, and I can give you an overview of what's what.

**Comment by feyeleanor** [comment]
I'll see if I can put my thoughts in sufficient good order to make sense :)

**Pull request merged**